src/utils/GovNotifyEvent.ts

```diff
@@ -32,17 +34,49 @@ export interface ReminderEmailEventDynamic {
 
 export const buildGovNotifyEventFields = (sessionId: string, yotiSessionId: string, personDetails: PersonIdentityItem): GovNotifyEvent => {
 	const nameParts = personIdentityUtils.getNames(personDetails);
-
-	return {
-		Message : {
-			sessionId,
-			yotiSessionId,
-			emailAddress: personIdentityUtils.getEmailAddress(personDetails),
-			firstName: nameParts.givenNames[0],
-			lastName: nameParts.familyNames[0],
-			messageType: Constants.PDF_EMAIL,
-		},
-	};
+	if (personDetails.pdfPreference === "letter") {
+		const addressesLength = personDetails.addresses.length;
+		return {
+			Message : {
+				sessionId,
+				yotiSessionId,
+				emailAddress: personIdentityUtils.getEmailAddress(personDetails),
+				firstName: nameParts.givenNames[0],
+				lastName: nameParts.familyNames[0],
+				messageType: Constants.PDF_EMAIL,
+				pdfPreference: personDetails.pdfPreference,
+				postalAddress: {
+					uprn: personDetails.addresses[addressesLength - 1].uprn,
```

Comment:
Are we sure that the addresses are in the order they are saved? the logic used here is based on the assumption that the last address in the array is the preferred address. Should we update the addresses with a new attribute like preferred address to identify the address to be used. Also good to know just by viewing the DB data?
cc @ViveakV

Reply:
What I was thinking was that when it comes time to send the user a posted letter then the lambda would query the database and if the user just has their send claims address then sen the letter to that, and if they've nominated a separate address then send it to that, as both would be the last item in the addresses array in each instance. I think adding an attribute for the postal address is a good idea just so it's clearer when viewing the data. What would be the preferable way to do this? Do we need to update the shared_claims address with a boolean called preferred_postal_address or something similar which is then either set to true or false in the front end, then if the user nominates a second address have that addresses' preferred_postal_address attribute set to true by default?

Reply:
@ViveakV

Reply:
What was discussed, can it be summarised here please.

Reply:
The solution implemented is to set a property on the initial user address that is brought over in shared_claims called preferredAddress to true by default in the F2FService mapAddresses function. Later if the user chooses another address for postal correspondence, this is sent to the BE DocumentSelection lambda and the saveUserPdfPreferences func in F2FService changes the user's existing address preferredAddress property to false  and the incoming second address will have the required property preferredAddress set to true, which is shown in the screenshot of the dynamoDB table on the PR

Reply:
Thanks for the details. I can see that its easier to view the preferred address in the DB now with this flag being set, but cant we get this address programmatically too by checking this flag instead of doing this - personDetails.addresses[addressesLength - 1]?

Reply:
Thanks for handling the request, code looks clean now.